controllers/machine_controller.go

```diff
@@ -376,38 +376,42 @@ func (r *MachineReconciler) isDeleteNodeAllowed(ctx context.Context, cluster *cl
 		return errNilNodeRef
 	}
 
-	// Get the control plane object associated with the cluster
-	controlPlane, err := getControlPlaneFromRef(ctx, r.Client, cluster.Spec.ControlPlaneRef)
-	if err != nil {
-		return err
+	var controlPlane *unstructured.Unstructured
+
+	// controlPlaneRef is an optional field in the Cluster so return nil if it isn't set
+	if cluster.Spec.ControlPlaneRef != nil {
+		var err error
+		controlPlane, err = external.Get(ctx, r.Client, cluster.Spec.ControlPlaneRef, cluster.Spec.ControlPlaneRef.Namespace)
+		if err != nil {
+			return err
+		}
 	}
 
 	// Check if the ControlPlane is externally managed (AKS, EKS, GKE, etc)
 	// and skip the following section if control plane is externally managed
 	// because there will be no control plane nodes registered
-	if !util.IsExternalManagedControlPlane(controlPlane) {
+	if controlPlane != nil && util.IsExternalManagedControlPlane(controlPlane) {
+		return nil
+	}
```

Comment:
If we move this after line 387, we can just check `util.IsExternalManagedControlPlane(controlPlane)` and return right away